core/gear_log/file_handle.ex

```diff
@@ -48,10 +48,14 @@ defmodule AntikytheraCore.GearLog.FileHandle do
   end
 
   defunp do_write(io_device :: :file.io_device, {time, level, context_id, msg} :: Message.t) :: :ok do
-    prefix = log_prefix(time, level, context_id)
-    formatted_lines_str = String.split(msg, "\n", trim: true) |> Enum.map_join(&(prefix <> &1 <> "\n"))
-    IO.write(io_device, formatted_lines_str)
-    write_debug_log(level, formatted_lines_str)
+    try do
+      prefix = log_prefix(time, level, context_id)
+      formatted_lines_str = String.split(msg, "\n", trim: true) |> Enum.map_join(&(prefix <> &1 <> "\n"))
+      IO.write(io_device, formatted_lines_str)
+      write_debug_log(level, formatted_lines_str)
+    rescue
+      _ in ErlangError -> write_debug_log(level, "The write process was skipped because it received malformed data.\n")
```

Comment:
- We should match against the reason part `:no_translation`.
In this case the rescued exception is `%ErlangError{original: :no_translation}`, and for this type of exceptions, exception wrapping introduced by `rescue` is (IMO) not very helpful.
We can fall back to the Erlang world by using `catch`.  
- We shouldn't enclose the whole code in `try`. We should enclose only code that may fail.
- When this error occurs we should put a message to the log file, not to tty (console).

So the resulting code should look like:
```ex
try do
  IO.write(io_device, formatted_lines_str)
catch
  :error, :no_translation -> IO.write(io_device, @message_about_malformed_log_message)
end
```

Reply:
fixed in 5045720

- The argument type and return value of `do_write/2` are changed. Because when crashing by receiving an malformed message, the process is already dead. Therefore, it is necessary to open the same log file, then update the handle.
- Also it seems beneficial to me to output error messages not only to the log file but also to the console.